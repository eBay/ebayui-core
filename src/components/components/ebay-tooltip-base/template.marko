<span
    w-on-expander-expand="handleExpand"
    w-on-expander-collapse="handleCollapse"
    w-body
    overlay-style=data.overlayStyle
    w-bind/>
