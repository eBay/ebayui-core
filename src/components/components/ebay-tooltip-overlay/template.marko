<script marko-init>
    var constants = require("./constants");
    var typeRoles = constants.typeRoles;
    var pointerStyles = constants.pointerStyles;
</script>

<var hasUserStyles=(
    data.styleTop ||
    data.styleLeft ||
    data.styleRight ||
    data.styleBottom
)/>
<var overlayStyles=(hasUserStyles ? {
    top: data.styleTop,
    left: data.styleLeft,
    right: data.styleRight,
    bottom: data.styleBottom
} : pointerStyles[data.pointer || "bottom"])/>

<div
    class="${data.type}__overlay"
    role=typeRoles[data.type]
    aria-labelledby=(data.type === "tourtip" && widget.elId("tourtip-label"))
    id=data.tooltipId
    style=overlayStyles
    w-bind>
    <span class="${data.type}__pointer ${data.type}__pointer--${data.pointer}"/>
    <div class="${data.type}__mask">
        <div class="${data.type}__cell">
            <div class="${data.type}__content">
                <if(data.heading)>
                    <h2
                        class=["${data.type}__heading", data.heading.class]
                        style=data.heading.style
                        w-id="tourtip-label"
                        w-body=data.heading.renderBody/>
                </if>

                <if(data.content)>
                    <div
                        body-only-if(!data.content.class && !data.content.style)
                        class=data.content.class
                        style=data.content.style
                        w-body=data.content.renderBody/>
                </if>
            </div>

            <if(data.type !== "tooltip")>
                <button
                    aria-label=data.a11yCloseText
                    class="${data.type}__close"
                    type="button"
                    w-on-click="handleCloseButton">
                    <span aria-hidden="true" role="img"/>
                </button>
            </if>
        </div>
    </div>
</div>
