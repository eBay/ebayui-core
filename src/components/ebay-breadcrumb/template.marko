<script marko-init>
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<nav w-bind
    aria-labelledby=widget.elId("breadcrumb-heading")
    class=["breadcrumb", data.class]
    style=data.style
    role="navigation"
    ${processHtmlAttributes(data)}>

    <${data.a11yHeadingTag || "h2"} id=widget.elId("breadcrumb-heading") class="clipped">
        ${data.a11yHeadingText}
    </>
    <ul>
        <li for(item in data.items | status-var="status")>
            <var current=(!item.href && status.isLast())/>
            <${item.href ? "a" : "button"}
                href=item.href
                aria-current=(current && "location")
                class=item.class
                style=item.style
                w-body=item.renderBody
                w-onclick=(!current && "handleClick")
                ${processHtmlAttributes(item)}/>
        </li>
    </ul>
</nav>
