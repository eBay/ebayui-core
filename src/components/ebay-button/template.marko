<script marko-init>
    var getWidgetId = require("../../common/get-marko-3-widget-id");
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<var size=(data.size || "medium")/>
<var priority=(data.priority || "secondary")/>
<var variant=(!data.variant && data.href ? "fake" : data.variant)/>
<var isIconVariant=(variant === 'icon')/>
<var isExpandVariant=(variant === 'expand')/>
<var isCtaVariant=(variant === 'cta')/>
<var isBadged=Boolean(data.badgeNumber && isIconVariant)/>
<var noText=(isIconVariant || isBadged || (isExpandVariant && data.noText))/>
<var baseClass=(variant ? "${variant}-btn" : "btn")/>
<var sizeClass="${baseClass}--${size}"/>
<var htmlAttributes=processHtmlAttributes(data)/>
<var tag=(data.href ? "a" : "button")/>

<${tag}
    ${htmlAttributes}
    w-bind
    w-onclick="handleClick"
    w-onkeydown="handleKeydown"
    id=(data.id || getWidgetId(out))
    class=[
        data.class,
        baseClass,
        noText && "${baseClass}--no-text",
        isBadged && "${baseClass}--badged",
        data.fluid && "${baseClass}--fluid",
        data.truncate && "${sizeClass}-truncated",
        data.fixedHeight && "${sizeClass}-fixed-height",
        !data.fixedHeight && !data.truncate && sizeClass,
        (priority === "secondary" || priority === "primary") && "${baseClass}--${priority}"
    ]
    style=data.style
    href=data.href
    type=(tag === "button" && data.type || "button")
    disabled=data.disabled
    aria-disabled=(data.partiallyDisabled && "true")>

    <if(isCtaVariant)>
        <span class="cta-btn__cell">
            <span w-body/>
            <ebay-icon type="inline" name="arrow-right-bold" class="cta-btn__icon" no-skin-classes/>
        </span>
    </if>
    <else>
        <var hasAriaLabel=Boolean(htmlAttributes.ariaLabel)/>
        <span body-only-if(!hasAriaLabel) w-body aria-hidden="true"/>
        <if(isBadged)>
            <ebay-badge
                number=data.badgeNumber
                type="icon"
                aria-label=(hasAriaLabel && data.badgeAriaLabel)
                aria-hidden=(hasAriaLabel && "true")/>
        </if>
    </else>
</>
