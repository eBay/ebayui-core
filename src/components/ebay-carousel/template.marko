<div class=data.classes style=data.style w-bind ${data.htmlAttributes}>
    <var config=data.config/>
    <var statusId=("carousel-status-" + widget.id)/>
    <div w-id="container" class=["carousel__container", data.bothControlsDisabled && "carousel__container--controls-disabled"]>
        <${data.a11yStatusTag}
            if(data.totalSlides >= 1)
            id=statusId
            class="clipped"
            aria-live=(data.autoplayInterval && !data.paused ? "off" : "polite")>
            <span>${data.a11yStatusText}</span>
        </>
        <button
            class="carousel__control carousel__control--prev"
            w-onclick=(!data.prevControlDisabled && "handleMove")
            data-direction=-1
            aria-describedby=statusId
            aria-label=data.a11yPreviousText
            aria-disabled=(data.prevControlDisabled && 'true')>
            <ebay-icon type="inline" name="chevron-left"/>
        </button>

        <ul w-id="list"
            class="carousel__list"
            style=(!config.nativeScrolling && data.offset && {
                transform:  "translate3d(" + (data.offset * -1) + "px,0,0)",
                transition: data.disableTransition ? 'none' : undefined
            })>
            <for(item in data.items)>
                <li ${item.htmlAttributes}
                    class=item.class
                    style=item.style
                    w-body=item.renderBody
                    aria-hidden=(!item.fullyVisible && 'true')
                    w-preserve-body-if(!!config.preserveItems)/>
            </for>
        </ul>

        <button
            w-id="next"
            class="carousel__control carousel__control--next"
            w-onclick=(!data.nextControlDisabled && "handleMove")
            data-direction=1
            aria-describedby=statusId
            aria-label=data.a11yNextText
            aria-disabled=(data.nextControlDisabled && 'true')>
            <ebay-icon type="inline" name="chevron-right"/>
        </button>

        <if(data.autoplayInterval && !data.bothControlsDisabled)>
            <if(data.paused)>
                <button
                    class="carousel__play"
                    aria-label=data.a11yPlayText
                    w-onclick="togglePlay">
                    <ebay-icon type="inline" name="play"/>
                </button>
            </if>
            <else>
                <button
                    class=`carousel__pause`
                    aria-label=data.a11yPauseText
                    w-onclick="togglePlay">
                    <ebay-icon type="inline" name="pause"/>
                </button>
            </else>
        </if>
    </div>

    <ul class="carousel__dots" if(!data.noDots && data.totalSlides > 1)>
        <li for(i from 0 to data.totalSlides - 1)>
            <var isActive=(i === data.slide)/>
            <button
                class={"carousel__dot--active": isActive}
                w-onclick=(!isActive && "handleDotClick")
                data-slide=i
                aria-describedby=statusId
                aria-disabled=(isActive && 'true')
                aria-label=(isActive ?
                    data.a11yCurrentText.replace('{currentSlide}', i + 1) :
                    data.a11yOtherText.replace('{slide}', i + 1))/>
        </li>
    </ul>
</div>
