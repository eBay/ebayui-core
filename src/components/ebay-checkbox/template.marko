<script marko-init>
    var processHtmlAttributes = require("../../common/html-attributes");
</script>

<span w-bind class=["checkbox", data.class] style=data.style>
    <input
        class="checkbox__control"
        type="checkbox"
        w-onchange="handleChange"
        disabled=data.disabled
        ${processHtmlAttributes(data)}/>
    <span class="checkbox__icon" hidden>
        <ebay-icon type="inline" name="checkbox-unchecked" class="checkbox__unchecked" focusable="false" no-skin-classes/>
        <ebay-icon type="inline" name="checkbox-checked" class="checkbox__checked" focusable="false" no-skin-classes/>
    </span>
</span>
