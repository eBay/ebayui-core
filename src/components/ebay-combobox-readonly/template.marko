<span
    class=data.class
    style=data.style
    ${data.htmlAttributes}
    w-on-expander-expand='handleExpand'
    w-on-expander-collapse='handleCollapse'
    w-preserve-attrs="class"
    w-bind>
    <span class=data.btnClass w-preserve-attrs="class">
        <input
            type='text'
            role="combobox"
            readonly
            value=(data.selectedOption && data.selectedOption.text)
            disabled=data.disabled
            aria-haspopup='listbox'
            aria-owns=widget.elId('combobox__options')
            aria-expanded='false'
            w-on-keydown='handleComboboxKeyDown'
            w-preserve-attrs="aria-expanded"
            ${data.htmlAttributes}/>
        <span class="combobox__icon" aria-hidden="true"></span>
    </span>
    <div
        if(data.options.length > 0)
        w-id='combobox__options'
        role='listbox'
        class=data.optionsClass>
        <div
            for(option in data.options)
            role='option'
            class=['combobox__option', option.class, (option.selected && 'combobox__option--active')]
            style=option.style
            ${option.htmlAttributes}
            tabindex="-1"
            w-on-click='handleOptionClick'
            w-preserve-attrs="tabindex,data-makeup-index"
            aria-selected=String(option.selected)
            data-option-value=option.value>
            <span>
                ${option.text}
            </span>
            <span class='combobox__status'/>
        </div>
    </div>
    <select
        hidden
        class='combobox__native'
        name=data.name>
        <option
            for(option in data.options)
            value=option.value
            selected=option.selected
            class=option.class>
        </option>
    </select>
</span>
