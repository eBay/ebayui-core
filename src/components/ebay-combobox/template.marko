<span
    class=data.class
    ${data.htmlAttributes}
    w-on-expander-expand='handleExpand'
    w-on-expander-collapse='handleCollapse'
    w-bind>
    <span class=data.btnClass w-preserve-attrs="class">
        <input
            type='text'
            value=(data.selectedOption && data.selectedOption.label)
            readonly
            disabled=data.disabled
            role='combobox'
            aria-haspopup='listbox'
            aria-owns=widget.elId('combobox__options')
            aria-expanded='false'
            w-on-keydown='handleComboboxKeyDown'
            w-preserve-attrs="aria-expanded"/>
        <span class="combobox__icon" aria-hidden="true"></span>
    </span>
    <div
        if(data.options.length > 0)
        w-id='combobox__options'
        role='listbox'
        class=data.optionsClass>
        <div
            for(option in data.options)
            role='option'
            class=['combobox__option', option.class]
            tabindex="-1"
            ${option.htmlAttributes}
            w-on-click='handleOptionClick'
            w-preserve-attrs="tabindex,data-makeup-index"
            aria-selected=String(option.selected)
            data-option-value=option.value>
            <span class='combobox__option-label'>
                ${option.label}
            </span>
            <span class='combobox__status'/>
        </div>
    </div>
    <select
        hidden
        class='combobox__native'
        name=data.name>
        <option
            for(option in data.options)
            value=option.value
            selected=option.selected
            class=option.class>
        </option>
    </select>
</span>
