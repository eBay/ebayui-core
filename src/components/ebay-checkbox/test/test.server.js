import { it, expect } from "vitest";

import { render } from "@marko/testing-library";
import template from "../index.marko";
import { testPassThroughAttributes } from "../../../common/test-utils/server";

it("renders default checkbox", async () => {
    const { getByRole } = await render(template);
    const checkbox = getByRole("checkbox");

    expect(checkbox).toMatchSnapshot();
    expect(checkbox.parentElement).toMatchSnapshot();
});

it("renders disabled checkbox", async () => {
    const { getByRole } = await render(template, { disabled: true });
    expect(getByRole("checkbox")).toMatchSnapshot();
});

it("renders checkbox with id", async () => {
    const { getByRole } = await render(template, { id: "abc123" });
    expect(getByRole("checkbox")).toMatchSnapshot();
});

testPassThroughAttributes(template, {
    getClassAndStyleEl(component) {
        return component.getByRole("checkbox").parentElement;
    },
});
