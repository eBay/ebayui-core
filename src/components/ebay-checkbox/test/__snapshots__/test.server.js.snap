// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`passes through additional html attributes 1`] = `
<input
  class="checkbox__control"
  data-marko="{"onclick":"forwardEvent s0 false 0","onfocus":"forwardEvent s0 false 1","onkeydown":"forwardEvent s0 false 2"}"
  data-passed-through="true"
  data-testid="AttributePassthrough"
  type="checkbox"
/>
`;

exports[`passes through additional html attributes 2`] = `
<span
  class="checkbox class1"
  style="color:red"
>
  <input
    class="checkbox__control"
    data-marko="{"onclick":"forwardEvent s0 false 0","onfocus":"forwardEvent s0 false 1","onkeydown":"forwardEvent s0 false 2"}"
    data-passed-through="true"
    data-testid="AttributePassthrough"
    type="checkbox"
  />
  <span
    class="checkbox__icon"
    hidden=""
  >
    <!--M#s0-5-0-->
    <svg
      aria-hidden="true"
      class="checkbox__unchecked"
      data-marko-key="@svg s0-5-0"
      focusable="false"
    >
      <defs
        data-marko-key="@defs s0-5-0"
      >
        <symbol
          id="icon-checkbox-unchecked-18"
          viewBox="0 0 18 18"
        >
          <path
            d="M6.4 17h5.2c1.137 0 1.929 0 2.546-.051.605-.05.953-.142 1.216-.276a3 3 0 0 0 1.311-1.311c.134-.263.226-.611.276-1.216.05-.617.051-1.41.051-2.546V6.4c0-1.137 0-1.929-.051-2.546-.05-.605-.142-.953-.276-1.216a3 3 0 0 0-1.311-1.311c-.263-.134-.611-.226-1.216-.276C13.529 1.001 12.736 1 11.6 1H6.4c-1.137 0-1.929 0-2.546.051-.605.05-.953.142-1.216.276a3 3 0 0 0-1.311 1.311c-.134.263-.226.611-.276 1.216C1.001 4.471 1 5.264 1 6.4v5.2c0 1.137 0 1.929.051 2.546.05.605.142.953.276 1.216a3 3 0 0 0 1.311 1.311c.263.134.611.226 1.216.276.617.05 1.41.051 2.546.051ZM.436 15.816C0 14.96 0 13.84 0 11.6V6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748Z"
          />
        </symbol>
      </defs>
      <use
        href="#icon-checkbox-unchecked-18"
      />
    </svg>
    <!--M/-->
    <!--M#s0-6-0-->
    <svg
      aria-hidden="true"
      class="checkbox__checked"
      data-marko-key="@svg s0-6-0"
      focusable="false"
    >
      <defs
        data-marko-key="@defs s0-6-0"
      >
        <symbol
          id="icon-checkbox-checked-18"
          viewBox="0 0 18 18"
        >
          <path
            d="M0 6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748C0 14.96 0 13.84 0 11.6V6.4Zm14.71-2.11a1 1 0 0 0-1.41 0L7 10.59l-2.29-2.3a1.004 1.004 0 0 0-1.42 1.42l3 3a1 1 0 0 0 1.41 0l7-7a1 1 0 0 0 .01-1.42Z"
          />
        </symbol>
      </defs>
      <use
        href="#icon-checkbox-checked-18"
      />
    </svg>
    <!--M/-->
  </span>
</span>
`;

exports[`renders checkbox with id 1`] = `
<input
  class="checkbox__control"
  data-marko="{"onclick":"forwardEvent s0 false 0","onfocus":"forwardEvent s0 false 1","onkeydown":"forwardEvent s0 false 2"}"
  id="abc123"
  type="checkbox"
/>
`;

exports[`renders default checkbox 1`] = `
<input
  class="checkbox__control"
  data-marko="{"onclick":"forwardEvent s0 false 0","onfocus":"forwardEvent s0 false 1","onkeydown":"forwardEvent s0 false 2"}"
  type="checkbox"
/>
`;

exports[`renders default checkbox 2`] = `
<span
  class="checkbox"
>
  <input
    class="checkbox__control"
    data-marko="{"onclick":"forwardEvent s0 false 0","onfocus":"forwardEvent s0 false 1","onkeydown":"forwardEvent s0 false 2"}"
    type="checkbox"
  />
  <span
    class="checkbox__icon"
    hidden=""
  >
    <!--M#s0-5-0-->
    <svg
      aria-hidden="true"
      class="checkbox__unchecked"
      data-marko-key="@svg s0-5-0"
      focusable="false"
    >
      <defs
        data-marko-key="@defs s0-5-0"
      >
        <symbol
          id="icon-checkbox-unchecked-18"
          viewBox="0 0 18 18"
        >
          <path
            d="M6.4 17h5.2c1.137 0 1.929 0 2.546-.051.605-.05.953-.142 1.216-.276a3 3 0 0 0 1.311-1.311c.134-.263.226-.611.276-1.216.05-.617.051-1.41.051-2.546V6.4c0-1.137 0-1.929-.051-2.546-.05-.605-.142-.953-.276-1.216a3 3 0 0 0-1.311-1.311c-.263-.134-.611-.226-1.216-.276C13.529 1.001 12.736 1 11.6 1H6.4c-1.137 0-1.929 0-2.546.051-.605.05-.953.142-1.216.276a3 3 0 0 0-1.311 1.311c-.134.263-.226.611-.276 1.216C1.001 4.471 1 5.264 1 6.4v5.2c0 1.137 0 1.929.051 2.546.05.605.142.953.276 1.216a3 3 0 0 0 1.311 1.311c.263.134.611.226 1.216.276.617.05 1.41.051 2.546.051ZM.436 15.816C0 14.96 0 13.84 0 11.6V6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748Z"
          />
        </symbol>
      </defs>
      <use
        href="#icon-checkbox-unchecked-18"
      />
    </svg>
    <!--M/-->
    <!--M#s0-6-0-->
    <svg
      aria-hidden="true"
      class="checkbox__checked"
      data-marko-key="@svg s0-6-0"
      focusable="false"
    >
      <defs
        data-marko-key="@defs s0-6-0"
      >
        <symbol
          id="icon-checkbox-checked-18"
          viewBox="0 0 18 18"
        >
          <path
            d="M0 6.4c0-2.24 0-3.36.436-4.216A4 4 0 0 1 2.184.436C3.04 0 4.16 0 6.4 0h5.2c2.24 0 3.36 0 4.216.436a4 4 0 0 1 1.748 1.748C18 3.04 18 4.16 18 6.4v5.2c0 2.24 0 3.36-.436 4.216a4 4 0 0 1-1.748 1.748C14.96 18 13.84 18 11.6 18H6.4c-2.24 0-3.36 0-4.216-.436a4 4 0 0 1-1.748-1.748C0 14.96 0 13.84 0 11.6V6.4Zm14.71-2.11a1 1 0 0 0-1.41 0L7 10.59l-2.29-2.3a1.004 1.004 0 0 0-1.42 1.42l3 3a1 1 0 0 0 1.41 0l7-7a1 1 0 0 0 .01-1.42Z"
          />
        </symbol>
      </defs>
      <use
        href="#icon-checkbox-checked-18"
      />
    </svg>
    <!--M/-->
  </span>
</span>
`;

exports[`renders disabled checkbox 1`] = `
<input
  class="checkbox__control"
  data-marko="{"onclick":"forwardEvent s0 false 0","onfocus":"forwardEvent s0 false 1","onkeydown":"forwardEvent s0 false 2"}"
  disabled=""
  type="checkbox"
/>
`;
